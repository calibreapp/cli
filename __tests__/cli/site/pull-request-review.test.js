import {
  runCLI,
  setupIntegrationServer,
  teardownIntegrationServer
} from '../../utils'

import getPullRequestReviewByBranch from '../../fixtures/getPullRequestReviewByBranch.json'

describe('display pull request review', () => {
  beforeAll(async () => await setupIntegrationServer(getPullRequestReviewByBranch))
  afterAll(async () => await teardownIntegrationServer())

  test('displays help', async () => {
    const out = await runCLI({
      args: 'site pull-request-review --help'
    })
    expect(out).toMatchSnapshot()
  })

  test('displays review', async () => {
    const out = await runCLI({
      args: 'site pull-request-review "author/my-pr" --site=test'
    })
    expect(out).toMatchSnapshot()
  })

  test('displays review json', async () => {
    const out = await runCLI({
      args: 'site pull-request-review "author/my-pr" --site=test --json'
    })
    expect(out).toMatchSnapshot()
  })

  test('displays review markdown', async () => {
    const out = await runCLI({
      args: 'site pull-request-review "author/my-pr" --site=test --markdown'
    })
    expect(out).toMatchSnapshot()
  })
})
