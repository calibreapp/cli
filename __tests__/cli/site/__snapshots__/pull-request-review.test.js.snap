// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display pull request review displays help 1`] = `
"calibre site pull-request-review <branch>

See the results of a Pull Request Review.

Options:

  --help      Show help                                                [boolean]
  --version   Show version number                                      [boolean]
  --site      A unique slug identifying each Site. You can find it in Site Setti
              ngs â†’ General or by using the calibre site list command.
                                                             [string] [required]
  --json      Outputs the results of the command in JSON format.
  --markdown  Outputs the results of the command in Markdown format."
`;

exports[`display pull request review displays review 1`] = `
"# Markdown here

"
`;

exports[`display pull request review displays review json 1`] = `
"{
  \\"title\\": \\"My Pull Request\\",
  \\"status\\": \\"completed\\",
  \\"branch\\": \\"author/branch-name\\",
  \\"markdownReport\\": \\"# Markdown here\\"
}"
`;

exports[`display pull request review displays review markdown 1`] = `"# Markdown here"`;
